app/services/support_interface/ministerial_report_export.rb

```diff
@@ -0,0 +1,52 @@
+module SupportInterface
+  class MinisterialReportExport
+    def call
+      results = subject_mapping(subject_status_count)
+
+      rows = {}
+
+      Subject::SUBJECTS.each do |subject|
+        rows[subject] = {
+          applications: 0,
+          offer_received: 0,
+          accepted: 0,
+          application_declined: 0,
+          application_rejected: 0,
+          application_withdrawn: 0,
+        }
+      end
+
+      results.each do |key, count|
+        subject, status = key
+
+        mapped_statuses = Subject::STATUS_MAPPING[status]
+
+        mapped_statuses.each do |mapped_status|
+          rows[:stem][mapped_status] += count if Subject::STEM_SUBJECTS.include? subject
+          rows[:ebacc][mapped_status] += count if Subject::EBACC_SUBJECTS.include? subject
+          rows[:secondary][mapped_status] += count if Subject::SECONDARY_SUBJECTS.include? subject
+          rows[subject][mapped_status] += count
+        end
+      end
+
+      rows[:total] = rows[:primary].merge(rows[:secondary]) { |_k, primary_value, secondary_value| primary_value + secondary_value }
```

Comment:
how does this work? Can merge take a block after? I've not seen it before

Reply:
yup!
![image](https://user-images.githubusercontent.com/47917431/134541256-7bd0df85-d6a4-4c66-bc63-411cb5edc991.png)
https://ruby-doc.org/core-3.0.2/Hash.html

Reply:
amazing. Thanks! That's awesome